src/diffusers/loaders.py

```diff
@@ -1495,55 +1520,33 @@ def load_lora_into_text_encoder(
-                    # Previously, the old LoRA layers were stored on the state dict at the
-                    # same level as the attention block i.e.
-                    # `text_model.encoder.layers.11.self_attn.to_out_lora.up.weight`.
-                    #
-                    # This is no actual module at that point, they were monkey patched on to the
-                    # existing module. We want to be able to load them via their actual state dict.
-                    # They're in `PatchedLoraProjection.lora_linear_layer` now.
-                    for name, _ in text_encoder_attn_modules(text_encoder):
-                        text_encoder_lora_state_dict[
-                            f"{name}.q_proj.lora_linear_layer.up.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_q_lora.up.weight")
-                        text_encoder_lora_state_dict[
-                            f"{name}.k_proj.lora_linear_layer.up.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_k_lora.up.weight")
-                        text_encoder_lora_state_dict[
-                            f"{name}.v_proj.lora_linear_layer.up.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_v_lora.up.weight")
-                        text_encoder_lora_state_dict[
-                            f"{name}.out_proj.lora_linear_layer.up.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_out_lora.up.weight")
-
-                        text_encoder_lora_state_dict[
-                            f"{name}.q_proj.lora_linear_layer.down.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_q_lora.down.weight")
-                        text_encoder_lora_state_dict[
-                            f"{name}.k_proj.lora_linear_layer.down.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_k_lora.down.weight")
-                        text_encoder_lora_state_dict[
-                            f"{name}.v_proj.lora_linear_layer.down.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_v_lora.down.weight")
-                        text_encoder_lora_state_dict[
-                            f"{name}.out_proj.lora_linear_layer.down.weight"
-                        ] = text_encoder_lora_state_dict.pop(f"{name}.to_out_lora.down.weight")
+                    attention_modules = text_encoder_attn_modules(text_encoder)
+                    text_encoder_lora_state_dict = convert_old_state_dict_to_peft(
+                        attention_modules, text_encoder_lora_state_dict
+                    )
+                # New diffusers format to PEFT
+                elif any("lora_linear_layer" in k for k in text_encoder_lora_state_dict.keys()):
+                    attention_modules = text_encoder_attn_modules(text_encoder)
+                    text_encoder_lora_state_dict = convert_diffusers_state_dict_to_peft(
```

Comment:
Yeah this part could be bit confusing for the maintainers to understand as to why we're using two different peft methods. Maybe if we name the previous one a bit more explicitly, it will help clear the confusion.

Reply:
Actually @younesbelkada @sayakpaul , I'd change the logic here slightly for now:

- 1. We convert all the different formats first to `diffusers` format (this way this method can be leveraged by the deprecated method). And then we convert all the differnt formats to peft.
So we should IMO have to methods:

```py
# 1. convert to diffusers format
text_encoder_lora_state_dict = convert_state_dict_to_diffusers(attention_modules, text_encoder_lora_state_dict)
# 2. convert to peft format
text_encoder_lora_state_dict = convert_diffusers_to_peft(attention_modules, text_encoder_lora_state_dict)
```

Then when saving we can add the reverse method:

```py
# 1. convert PEFT to diffusers:
text_encoder_lora_state_dict = convert_peft_to_diffusers(attention_modules, text_encoder_lora_state_dict)
```

This has some advantages:
- 1.) Easier to read
- 2.) Easier to maintain / test
- 3.) Easier to add new conversions

Reply:
Proposed something in [f8e87f6](https://github.com/huggingface/diffusers/pull/5058/commits/f8e87f6220a3ae2c63119e4f83f23f8cc321cbce) , we could easily extend that to other formats if needed. Let me know what do you think